x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportTriggerTransformAction.java

```diff
@@ -0,0 +1,177 @@
+        this.threadPool = threadPool;
+        this.securityContext = XPackSettings.SECURITY_ENABLED.get(settings)
+            ? new SecurityContext(settings, threadPool.getThreadContext())
+            : null;
+    }
+
+    @Override
+    protected void doExecute(Task task, Request request, ActionListener<Response> listener) {
+        final ClusterState clusterState = clusterService.state();
+        XPackPlugin.checkReadyForXPackCustomMetadata(clusterState);
+
+        final DiscoveryNodes nodes = clusterState.nodes();
+        if (nodes.isLocalNodeElectedMaster() == false) {
+            // Delegates trigger transform to elected master node so it becomes the coordinating node.
+            if (nodes.getMasterNode() == null) {
+                listener.onFailure(new MasterNotDiscoveredException());
+            } else {
+                transportService.sendRequest(
+                    nodes.getMasterNode(),
+                    actionName,
+                    request,
+                    new ActionListenerResponseHandler<>(listener, Response::new)
+                );
+            }
+            return;
+        }
+        useSecondaryAuthIfAvailable(securityContext, () -> {
+            // set headers to run transform as calling user
+            Map<String, String> filteredHeaders = ClientHelper.getPersistableSafeSecurityHeaders(
+                threadPool.getThreadContext(),
+                clusterService.state()
+            );
+
+            PersistentTasksCustomMetadata.PersistentTask<?> transformTask = TransformTask.getTransformTask(request.getId(), clusterState);
+
+            // to send a request to trigger the transform at runtime, several requirements must be met:
+            // - transform must be running, meaning a task exists
+            // - transform is not failed (stopped transforms do not have a task)
+            if (transformTask != null
+                && transformTask.isAssigned()
+                && transformTask.getState() instanceof TransformState
+                && ((TransformState) transformTask.getState()).getTaskState() != TransformTaskState.FAILED) {
+
+                ActionListener<Response> taskTriggerListener = ActionListener.wrap(listener::onResponse, e -> {
+                    // benign: A transform might have been stopped meanwhile, this is not a problem
+                    if (e instanceof TransformTaskDisappearedDuringTriggerException) {
+                        logger.debug("[{}] transform task disappeared during trigger, ignoring", request.getId());
+                        listener.onResponse(Response.TRUE);
+                        return;
+                    }
+                    if (e instanceof TransformTaskTriggerException) {
+                        logger.warn(() -> format("[%s] failed to trigger running transform.", request.getId()), e);
+                        listener.onResponse(Response.TRUE);
+                        return;
+                    }
```

Comment:
in a mixed version cluster with old node versions you could get a `ActionNotFoundException`.

(alternatively you can filter/check the node version before `request.setNodes`, however `ActionNotFound` seems simpler, especially as this problem will eventually disappear)

Reply:
it should of course error. 

What error will be returned with letting it fall through?

Reply:
`POST _transform/test-from-1/_trigger`

errors the following way:
```
{
  "error": "no handler found for uri [/_transform/test-from-1/_trigger?pretty=true] and method [POST]"
}
```

This seems good to me.

Reply:
Actually I was too quick answering this comment.
FTR, I investigated more closely and here is what happens:
Setup:
2 nodes:
- 1 node (`localhost:9200`) running of `main` so it **doesn't** contain `_trigger` API
- 1 node (`localhost:9201`) running of this PR so it contains `_trigger` API
2 transforms:
- continuous transform `test-from-1` with 1 task on node `localhost:9201` (with `_trigger`)
- continuous transform `test-from-2` with 1 task on node `localhost:9200` (**without** `_trigger`)

Scenario:
```
elasticsearch-8.7.0-SNAPSHOT []$ curl -k -u elastic:mMEyCrC++MoB3_kaESK+ -X POST https://localhost:9200/_transform/test-from-1/_trigger
{"error":"no handler found for uri [/_transform/test-from-1/_trigger] and method [POST]"}%                                                                                                                  

elasticsearch-8.7.0-SNAPSHOT []$ curl -k -u elastic:mMEyCrC++MoB3_kaESK+ -X POST https://localhost:9200/_transform/test-from-2/_trigger
{"error":"no handler found for uri [/_transform/test-from-2/_trigger] and method [POST]"}%                                                                                                                  

elasticsearch-8.7.0-SNAPSHOT []$ curl -k -u elastic:mMEyCrC++MoB3_kaESK+ -X POST https://localhost:9201/_transform/test-from-1/_trigger
{"acknowledged":true}%                                                                                                                                                                                      

elasticsearch-8.7.0-SNAPSHOT []$ curl -k -u elastic:mMEyCrC++MoB3_kaESK+ -X POST https://localhost:9201/_transform/test-from-2/_trigger
{"acknowledged":true}%                                                                                                                                                                                      
```

The first 2 are obvious, we try to call the `_trigger` API on a node that doesn't have it, so the error is returned.
The 3. is ok. We call `_trigger` API on the node that has it, the task is on the same node so it works fine.
The 4. is not ok. We call `_trigger` API on the node that has it, but the task is on a different node.
The exception is swallowed, in the logs we see:
```
[2023-01-30T13:11:42,238][WARN ][o.e.x.t.a.TransportTriggerTransformAction] [node-trigger] [test-from-2] failed to trigger running transform.org.elasticsearch.xpack.transform.action.TransportTriggerTransformAction$TransformTaskTriggerException: Failed to trigger running transform.
	at org.elasticsearch.xpack.transform.action.TransportTriggerTransformAction.newResponse(TransportTriggerTransformAction.java:130)
	at org.elasticsearch.xpack.transform.action.TransportTriggerTransformAction.newResponse(TransportTriggerTransformAction.java:43)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction.newResponse(TransportTasksAction.java:214)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.finishHim(TransportTasksAction.java:325)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onFailure(TransportTasksAction.java:315)
```

Let me fix that...

Reply:
Ok, now the `ActionNotFoundTransportException` exception is a fall-through exception with the following API error message:
```
{
  "error": {
    "root_cause": [
      {
        "type": "action_not_found_transport_exception",
        "reason": "No handler for action [cluster:admin/transform/trigger[n]]"
      }
    ],
    "type": "action_not_found_transport_exception",
    "reason": "No handler for action [cluster:admin/transform/trigger[n]]"
  },
  "status": 500
}
```

Additionally, I made the `_trigger` API fail if the transform (config, **not** task) does not exist.